### PR TITLE
Newtype, variants and unit

### PR DESCRIPTION
**Pull request by MabezDev**
Adds ser/de for the following types:

* `unit` types, i.e `()` & `struct Test;` 
* `newtype` structs, ser/de'ing the underlying value
* Variants:
    - The `newtype_variant` format, i.e `Variant(u32)`
    - The  `struct_variant` format, i.e `Variant { x: u32, y: u16 }`

### CODE REVIEW

**Comment by MathiasKoch** [comment]
+1 on this! Did a test yesterday, and it also fixes support for externally tagged enums (https://serde.rs/enum-representations.html)

Good job!

**Comment by MathiasKoch** [comment]
Ping @japaric

File: src/ser/struct_.rs [hunk 1]
```diff
@@ -52,3 +52,52 @@ where
...
+    B: ArrayLength<u8>,
+{
+    pub(crate) fn new(ser: &'a mut Serializer<B>) -> Self {
+        SerializeStructVariant { ser, first: true }
+    }
+}
+
+impl<'a, B> ser::SerializeStructVariant for SerializeStructVariant<'a, B>
+where
+    B: ArrayLength<u8>,
```
Comment (eldruin) [inline src/ser/struct_.rs:75]: This will need to be adapted once #36 lands

**Review by eldruin — COMMENTED** [review]
Other than the merging with #36 (or #27?), looks good to me.
An entry to the changelog would also be necessary, but maybe we can add all those entries in a dedicated PR after merging all of the currently pending ones.

**Comment by eldruin** [comment]
@MathiasKoch how do you want to go about this one?
Would you be able to pull these changes into a new PR and adapt them?

**Comment by MathiasKoch** [comment]
@MabezDev Thank you very much for your work here!

I am closing this, as i have moved it to #42 with a rebase on master, to get this moving.

**Pull request closed without merging**